packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_function_calling.dart

```diff
@@ -21,7 +21,13 @@ import 'vertex_schema.dart';
 /// knowledge and scope of the model.
 final class Tool {
   /// Constructor
-  Tool({this.functionDeclarations});
+  Tool._({this.functionDeclarations});
+
+  /// Returns a [Tool] instance with list of [FunctionDeclaration].
+  static Tool functionDeclarationsTool(
```

Comment:
I don't think it is useful to repeate `Tool` in the name.

how about `static Tool functionDeclarations(`

Reply:
It's same with the attributes. Would you suggest to make the attribute functionDeclarations private?

Reply:
Hmm, good point. I guess I might skip the static method altogether in that case. It's not really saving any characters compared to the constructor and named argument.